src/test/java/teammates/sqlui/webapi/CreateInstructorActionTest.java

```diff
@@ -0,0 +1,256 @@
+                null, false);
+
+        when(mockLogic.getCourse(typicalCourse.getId())).thenReturn(typicalCourse);
+        when(mockLogic.createInstructor(any(Instructor.class))).thenReturn(newInstructor);
+
+        loginAsAdmin();
+
+        CreateInstructorAction action = getAction(requestBody, params);
+        JsonResult r = getJsonResult(action);
+        InstructorData response = (InstructorData) r.getOutput();
+
+        verifySpecifiedTasksAdded(Const.TaskQueue.INSTRUCTOR_COURSE_JOIN_EMAIL_QUEUE_NAME, 1);
+        verifySpecifiedTasksAdded(Const.TaskQueue.SEARCH_INDEXING_QUEUE_NAME, 1);
+
+        TaskWrapper taskAdded = mockTaskQueuer.getTasksAdded().get(0);
+        assertEquals(typicalCourse.getId(), taskAdded.getParamMap().get(Const.ParamsNames.COURSE_ID));
+        assertEquals(newInstructor.getEmail(), taskAdded.getParamMap().get(Const.ParamsNames.INSTRUCTOR_EMAIL));
+
+        assertEquals(newInstructor.getName(), response.getName());
+        assertEquals(newInstructor.getEmail(), response.getEmail());
+    }
+
+    @Test
+    void testAccessControl_noLogin_cannotAccess() {
+        String[] params = new String[] {
+                Const.ParamsNames.COURSE_ID, typicalCourse.getId(),
+        };
+
+        logoutUser();
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testAccessControl_unregisteredUsers_cannotAccess() {
+        String[] params = new String[] {
+                Const.ParamsNames.COURSE_ID, typicalCourse.getId(),
+        };
+
+        loginAsUnregistered("unregistered user");
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testAccessControl_students_cannotAccess() {
+        Student typicalStudent = getTypicalStudent();
+        String[] params = new String[] {
+                Const.ParamsNames.COURSE_ID, typicalCourse.getId(),
+        };
+
+        loginAsStudent(typicalStudent.getGoogleId());
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testAccessControl_instructorsFromDifferentCourse_cannotAccess() {
+        Course differentCourse = new Course("different id", "different name", Const.DEFAULT_TIME_ZONE,
+                "teammates");
+        Instructor instructorFromDifferentCourse = getTypicalInstructor();
+        instructorFromDifferentCourse.setCourse(differentCourse);
+        instructorFromDifferentCourse.setGoogleId("different google id");
+
+        String[] params = new String[] {
+                Const.ParamsNames.COURSE_ID, typicalCourse.getId(),
+        };
+
+        when(mockLogic.getCourse(typicalCourse.getId())).thenReturn(typicalCourse);
+        when(mockLogic.getInstructorByGoogleId(differentCourse.getId(), instructorFromDifferentCourse.getGoogleId()))
+                .thenReturn(instructorFromDifferentCourse);
+
+        loginAsInstructor(instructorFromDifferentCourse.getGoogleId());
+
+        verifyCannotAccess(params);
+    }
+
+    @Test
+    void testAccessControl_instructorWithoutCorrectCoursePrivilege_cannotAccess() {
```

Comment:
Could you also add a positive test case to cover this? i.e. can access with correct privileges